src/main/java/reposense/authorship/analyzer/AnnotatorAnalyzer.java

```diff
@@ -53,15 +52,23 @@ public static void aggregateAnnotationAuthorInfo(FileInfo fileInfo, Map<String,
      *         {@code Optional.empty()} if an end author tag is used (i.e. "@@author"),
      *         or if the extracted author name is too short.
      */
-    private static Optional<Author> findAuthorInLine(String line, Map<String, Author> authorAliasMap) {
+    private static Optional<Author> findAuthorInLine(String line, Map<String, Author> authorAliasMap,
+                                                     Optional<Author> currentAnnotatedAuthor) {
         try {
             String[] split = line.split(AUTHOR_TAG);
             String name = extractAuthorName(split[1]);
             if (name == null) {
+                if (!currentAnnotatedAuthor.isPresent()) {
+                    // Attribute to unknown author if an empty author tag was provided, but not as an end author tag
+                    return Optional.of(Author.UNKNOWN_AUTHOR);
+                }
                 return Optional.empty();
             }
             return Optional.of(authorAliasMap.getOrDefault(name, Author.UNKNOWN_AUTHOR));
         } catch (ArrayIndexOutOfBoundsException e) {
+            if (!currentAnnotatedAuthor.isPresent()) {
+                return Optional.of(Author.UNKNOWN_AUTHOR);
+            }
             return Optional.empty();
```

Comment:
for future reference, this may be a good place to use the ternary operator.

```java
return currentAnnotatedAuthor.isPresent() ? Optional.empty() : Optional.of(Author.UNKNOWN_AUTHOR);
```

Reply:
Yes, truly. However, many languages other than Java do not support ternary operators.